packages/ra-ui-materialui/src/KeyboardShortcut.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import {
+    ComponentsOverrides,
+    styled,
+    SxProps,
+    Typography,
+} from '@mui/material';
+import clsx from 'clsx';
+
+export const KeyboardShortcut = ({
+    className,
+    keyboardShortcut,
+    ...rest
+}: KeyboardShortcutProps) => {
+    if (!keyboardShortcut) {
+        return null;
+    }
+
+    return (
+        <Root
+            className={clsx(KeyboardShortcutClasses.root, className)}
+            {...rest}
+        >
+            {keyboardShortcut
+                .split('>')
+                .map((sequence, sequenceIndex, sequences) => (
+                    <React.Fragment key={`${sequence}-${sequenceIndex}`}>
+                        {sequence.split('+').map((key, keyIndex) => (
+                            <React.Fragment key={`${key}-${keyIndex}`}>
+                                <Typography
+                                    component="kbd"
+                                    className={KeyboardShortcutClasses.kbd}
+                                    key={key}
+                                >
+                                    {KeyMap[key]
+                                        ? KeyMap[key]
+                                        : key.toUpperCase()}
+                                </Typography>
+                            </React.Fragment>
+                        ))}
+                        {sequenceIndex < sequences.length - 1 ? (
+                            <>&nbsp;</>
+                        ) : null}
+                    </React.Fragment>
+                ))}
+        </Root>
+    );
+};
+
+const KeyMap = {
+    meta: '⌘',
+    ctrl: '⌃',
+    shift: '⇧',
+    alt: '⌥',
+    enter: '⏎',
+    escape: '⎋',
+    backspace: '⌫',
+    delete: '⌦',
+    tab: '⇥',
+    space: '␣',
+    up: '↑',
+    down: '↓',
+    left: '←',
+    right: '→',
+    home: '↖',
+    end: '↘',
+    pageup: '⇞',
+    pagedown: '⇟',
+};
+
+export interface KeyboardShortcutProps
+    extends React.DetailedHTMLProps<
+        React.HTMLAttributes<HTMLDivElement>,
+        HTMLDivElement
+    > {
+    keyboardShortcut?: string;
+    sx?: SxProps;
+}
+
+const PREFIX = 'RaKeyboardShortcut';
+const KeyboardShortcutClasses = {
+    root: `${PREFIX}-root`,
+    kbd: `${PREFIX}-kbd`,
+};
+
+const Root = styled('div')(({ theme }) => ({
```

Comment:
by default, I would dim the keyboard shortcut so that it's less visible than the title. Maybe with opacity?